runners/google-cloud-dataflow-java/worker/src/main/java/org/apache/beam/runners/dataflow/worker/BatchDataflowWorker.java

```diff
@@ -370,6 +370,8 @@ boolean doWork(WorkItem workItem, WorkItemStatusClient workItemStatusClient) thr
                 counterSet,
                 executionContext,
                 stageName);
+      } else {
+        throw new IllegalStateException("Work Item was neither a MapTask nor a SourceOperation");
```

Comment:
This is a change in the default behavior (throwing vs passing `null` worker). Please verify that this change is intended / correct.